mammoth/trainer.py

```diff
@@ -496,10 +516,13 @@ def _gradient_accumulation(
                 total_stats.update(batch_stats)
                 report_stats.update(batch_stats)
                 report_stats.update_task_loss(batch_stats.loss, metadata)
-
             except Exception:
```

Comment:
i think this is also meant to to catch cuda oom's, isn't it? in which case your nan counter isn't valid?

Reply:
Yeah, that's right. The current code maybe allows recovering from a single slightly too large batch, by catching the OOM. This is quite unlikely, though, and wasn't intentional.

We could have a typed exception that increases the nan counter, and let everything else through directly. That would cause OOMs to be instantly fatal.